api/src/main/java/org/openmrs/Cohort.java

```diff
@@ -144,44 +149,46 @@ public Cohort(String commaSeparatedIds) {
 	 */
 	public String getCommaSeparatedPatientIds() {
 		StringBuilder sb = new StringBuilder();
-		for (Iterator<Integer> i = getMemberIds().iterator(); i.hasNext();) {
-			sb.append(i.next());
-			if (i.hasNext()) {
-				sb.append(",");
-			}
+        for (CohortMember member : getMembers()) {
+			sb.append(member.getPatientId().getPatientId().toString());
+			sb.append(",");
 		}
-		return sb.toString();
+		return sb.length() > 0 ? sb.substring(0, sb.length() - 1) : "";
 	}
 
 	public boolean contains(Patient patient) {
-		return getMemberIds() != null && getMemberIds().contains(patient.getPatientId());
+		return getMembers() != null && getMembers().stream()
```

Comment:
Probably we should add a TODO item here that this needs to be changed to compare the Patient objects instead of their ids when we finally remove the deprecated methods that take ids